ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractMinerExecutor.java

```diff
@@ -60,10 +69,32 @@ public AbstractMinerExecutor(
     this.gasLimitCalculator = gasLimitCalculator;
   }
 
-  public abstract M startAsyncMining(
-      final Subscribers<MinedBlockObserver> observers, final BlockHeader parentHeader);
+  public Optional<M> startAsyncMining(
+      final Subscribers<MinedBlockObserver> observers, final BlockHeader parentHeader) {
+    try {
+      final M currentRunningMiner = createMiner(observers, parentHeader);
+      executorService.execute(currentRunningMiner);
+      return Optional.of(currentRunningMiner);
+    } catch (RejectedExecutionException e) {
+      LOG.warn("Unable to start mining.", e);
+      return Optional.empty();
+    }
+  }
+
+  public void shutDown() {
+    if (stopped.compareAndSet(false, true)) {
+      executorService.shutdownNow();
+    }
+  }
+
+  public void awaitShutdown() throws InterruptedException {
+    if (!executorService.awaitTermination(30, TimeUnit.SECONDS)) {
+      LOG.error("Failed to shutdown {}.", this.getClass().getSimpleName());
+    }
+  }
 
-  public abstract M createMiner(final BlockHeader parentHeader);
+  public abstract M createMiner(
```

Comment:
nit: I know this is from prior - but having an executor class providing a "createMiner" public function feels odd i.e. this class was made to run things, not make things - maybe a rename can clean it up?